src/SearchBar.jsx

```diff
@@ -0,0 +1,201 @@
+var React           = require('react')
+	, Popup           = require('./Popup')
+	, activeElement   = require('react/lib/getActiveElement')
+	, List 						= require('./List')
+	, compat          = require('./util/compat')
+	, _               = require('./util/_')
+	, cx 							= require('classnames')
+	, PlainList       = require('./List')
+	, createUncontrolledWidget = require('uncontrollable');
+
+var SearchBar = React.createClass({
+
+	mixins: [
+		require('./mixins/WidgetMixin'),
+		require('./mixins/TimeoutMixin'),
+		require('./mixins/PureRenderMixin'),
+		require('./mixins/DataFilterMixin'),
+		require('./mixins/DataHelpersMixin'),
+		require('./mixins/PopupScrollToMixin'),
+		require('./mixins/RtlParentContextMixin'),
+		require('./mixins/ListMovementMixin')
+	],
+
+	getDefaultProps(){
+		return{
+			className:'',
+			open:false,
+			disabled:false,
+			readOnly:false,
+			busy:false,
+			dropUp:false,
+			messages: msgs()
+		}
+	},
+
+	getInitialState: function(){
+		const data = this.props.data
+			, initialIdx = this._dataIndexOf(data, this.props.value);
+
+
+		return {
+			selected:data[initialIdx],
+			focused:data[initialIdx] || data[0],
+			search:'',
```

Comment:
I would move search to props, and change the name to `searchTerm` to be consistent with the multiselect. the `createUncontrolledWidget` function will automatically wrap the the SearchBar so that if a user doesn't want to manage the searchTerm prop themselves the widget will do it for them (essentially acting like its in state)

basically any place you'd want to setState for `search` just `notify('onSearch')` instead